lib/ImageProcessor.js

```diff
@@ -51,18 +51,21 @@ class ImageProcessor {
         const bucket = config.get("bucket");
         const jpegOptimizer = config.get("jpegOptimizer", "mozjpeg");
 
-        let promise = new Promise((resolve) => { resolve(); });
+        let promise = Promise.resolve();
         let processedImages = 0;
+        let removeAfterSuccess = false;
 
         if ( config.exists("backup") ) {
             const backup = config.get("backup");
             backup.acl = backup.acl || acl;
             backup.bucket = backup.bucket || bucket;
+            // If backup.move option is true, remove original object after completion
```

Comment:
(hint) I'm not a huge fan of in-code comments. In most cases they seems to be needed, because functionality or variable/method names are not self-explanatory :)